src/NUnit.Portable.Agent/NUnitPortableDriver.cs

```diff
@@ -114,18 +119,23 @@ public string Run(Action<string> callback, string filter)
         /// <param name="filter">A filter that controls which tests are executed</param>
         public void RunAsync(Action<string> callback, string filter)
         {
-            CheckLoadWasCalled();
-            log.Info("Running {0} - see separate log file", _testAssembly.FullName);
-            ExecuteMethod(RUN_ASYNC_METHOD, new[] { typeof(Action<string>), typeof(string) }, callback, filter);
+            CheckAssembliesLoaded();
+
+            foreach (var package in _testPackages)
+            {
+                log.Info("Running {0} - see separate log file", package.FullName);
+                package.ExecuteMethod(RUN_ASYNC_METHOD, new[] { typeof(Action<string>), typeof(string) }, callback, filter);
+            }
         }
 
```

Comment:
Now I've typed out the words 'hijack the callback', that seems like almost definitely the solution...